tensorflow/python/autograph/converters/loop_integration_test.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Integration Tests for loop."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.autograph.converters import break_statements
+from tensorflow.python.autograph.converters import continue_statements
+from tensorflow.python.autograph.converters import control_flow
+from tensorflow.python.autograph.core import converter_testing
+from tensorflow.python.framework import constant_op
+from tensorflow.python.platform import test
+
+
+class LoopIntegrationTest(converter_testing.TestCase):
+
+  def assertTransformedEquivalent(self, test_fn, *inputs):
+    with self.converted(test_fn, [break_statements,
+                                  continue_statements,
+                                  control_flow],
+                        {}, (constant_op.constant,)) as result:
+      self.assertEqual(test_fn(*inputs), result.test_fn(*inputs))
+
+  def test_while_loop(self):
+
+    def test_fn(x):
+      v = []
+      while x > 0:
+        x -= 1
+        if x % 2 == 0:
+          break
+        v.append(x)
+      return v
+
+    self.assertTransformedEquivalent(test_fn, 0)
+    self.assertTransformedEquivalent(test_fn, 1)
+    self.assertTransformedEquivalent(test_fn, 4)
+
+  def test_for_loop(self):
+
+    def test_fn(a):
+      v = []
+      for x in a:
+        x -= 1
+        if x % 2 == 0:
+          break
+        v.append(x)
+      return v
+
+    with self.converted(test_fn, break_statements, {},
+                        (constant_op.constant,)) as result:
+      # The break is incompletely canonicalized. The loop will not interrupt,
+      # but the section following the break will be skipped.
```

Comment:
I think the PR is meant to fix exactly this issue, is it not?

Reply:
I'm really sorry about this... I just copied the test from break_statement_test.py for the reason I mentioned above and didn't notice the extra self.converted... Anyway, those two unnecessary tests are removed...

Reply:
Ah, I see. It strikes me that break_statement_test.py now has a comment that is stale - feel free to update it as well.